integration-tests/soak/ocr_test.go

```diff
@@ -22,58 +30,183 @@ func TestOCRSoak(t *testing.T) {
 func TestOCRSoak_ChangeBlockGasLimit(t *testing.T) {
 	config, err := tc.GetConfig(t.Name(), tc.OCR)
 	require.NoError(t, err, "Error getting config")
-	runOCRSoakTest(t, config, "")
+	ocrSoakTest, err := testsetups.NewOCRSoakTest(t, &config, false)
+	require.NoError(t, err, "Error creating OCR soak test")
+	runOCRSoakTest(t, ocrSoakTest, config, "")
+}
+
+// TestOCRSoak_RPCDownForAllCLNodes simulates a network chaos by bringing down network to RPC node for all Chainlink Nodes
+func TestOCRSoak_RPCDownForAllCLNodes(t *testing.T) {
+	config, err := tc.GetConfig(t.Name(), tc.OCR)
+	require.NoError(t, err, "Error getting config")
+	require.True(t, config.Network.IsSimulatedGethSelected(), "This test requires simulated geth")
+
+	namespace := fmt.Sprintf("%s-%s", "soak-ocr-simulated-geth", uuid.NewString()[0:5])
+	chaos, err := gethNetworkDownChaos(GethNetworkDownChaosOpts{
+		DelayCreate: time.Minute * 2,
+		Duration:    time.Minute * 5,
+		Name:        "ocr-soak-test-geth-network-down-all-nodes",
+		Description: "Geth Network Down For All Chainlink Nodes",
+		Namespace:   namespace,
+		TargetSelector: v1alpha1.PodSelector{
+			Selector: v1alpha1.PodSelectorSpec{
+				GenericSelectorSpec: v1alpha1.GenericSelectorSpec{
+					Namespaces:     []string{namespace},
+					LabelSelectors: map[string]string{"app": "chainlink-0"},
+				},
+			},
+			Mode: v1alpha1.AllMode,
+		},
+	})
+	require.NoError(t, err, "Error creating chaos")
+	ocrSoakTest, err := testsetups.NewOCRSoakTest(t, &config, false,
+		testsetups.WithNamespace(namespace),
+		testsetups.WithChaos([]*k8schaos.Chaos{chaos}),
+	)
+	require.NoError(t, err, "Error creating OCR soak test")
+	runOCRSoakTest(t, ocrSoakTest, config, "")
 }
 
-func runOCRSoakTest(t *testing.T, config tc.TestConfig, customNetworkTOML string) {
+// TestOCRSoak_RPCDownForAllCLNodes simulates a network chaos by bringing down network to RPC node for 50% of Chainlink Nodes
+func TestOCRSoak_RPCDownForHalfCLNodes(t *testing.T) {
+	config, err := tc.GetConfig(t.Name(), tc.OCR)
+	require.NoError(t, err, "Error getting config")
+	require.True(t, config.Network.IsSimulatedGethSelected(), "This test requires simulated geth")
+
+	namespace := fmt.Sprintf("%s-%s", "soak-ocr-simulated-geth", uuid.NewString()[0:5])
+	chaos, err := gethNetworkDownChaos(GethNetworkDownChaosOpts{
+		DelayCreate: time.Minute * 2,
+		Duration:    time.Minute * 5,
+		Name:        "ocr-soak-test-geth-network-down-half-nodes",
+		Description: "Geth Network Down For 50 Percent Of Chainlink Nodes",
+		Namespace:   namespace,
+		TargetSelector: v1alpha1.PodSelector{
+			Selector: v1alpha1.PodSelectorSpec{
+				GenericSelectorSpec: v1alpha1.GenericSelectorSpec{
+					Namespaces:     []string{namespace},
+					LabelSelectors: map[string]string{"app": "chainlink-0"},
+				},
+			},
+			Mode:  v1alpha1.FixedPercentMode,
+			Value: "50",
+		},
+	})
```

Comment:
this and previous chaos options differ only in 2 values, why not extract common part?

Reply:
@Tofel it's not only these values. The common part is extracted in gethNetworkDownChaos(). Keeping `GethNetworkDownChaosOpts` in the tests makes them more easy to understand. E.g. how long chaos will take, etc.